libipl/p10/ipl0.C

```diff
@@ -344,6 +344,53 @@ static void process_guard_records()
 	}
 }
 
+/*
+ * @Brief Function will check if the FCO state is set for hardware units
+ * consumed by sbe or not. If FCO override bit is set for such hardware
+ * units, need to mark present and functional state as set in devicetree
+ * during the boot.
+ *
+ */
+static void apply_fco_override(void)
+{
+	std::array<const char*, 8> mProcChild =
+		{"core", "pauc", "pau", "iohs", "mc", "chiplet", "pec", "fc"};
+
+	struct pdbg_target *proc, *child;
+	uint8_t buf[5];
+
+	pdbg_for_each_class_target("proc", proc) {
+		if(!ipl_is_master_proc(proc))
+			continue;
+
+		for(const char* data : mProcChild) {
+			pdbg_for_each_target(data, proc, child) {
+
+				if (!pdbg_target_get_attribute_packed(child,
+							"ATTR_HWAS_STATE",
+							"41", 1, buf)) {
+					ipl_error_callback(IPL_ERR_ATTR_READ_FAIL);
+				}
+
+				//0-1 bits reserved
+				//2nd bit Functional override
+				//3rd bit spec deconfig
+				//4th bit Dump capable
+				//5th bit Functional
+				//6th bit Present
+				//7th bit Deconfig
+				//Checking the FCO bit is set or not. If it's set means
+				//functional and present state of target should be set.
+				if(buf[4] & 0x04) {
```

Comment:
You don't have access to a structure?

Reply:
ipl , isn not linked with lib-dt api's which is used api based access.